apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/sink/direct/MbiCache.java

```diff
@@ -159,37 +184,53 @@ static class DatabaseLookupFunction implements LookupFunction {
      */
     @VisibleForTesting
     @Override
-    public Mbi lookupMbi(String mbi) {
+    public Mbi lookupMbiImpl(String mbi) {
+      Mbi result = null;
+      int loopNumber = 0;
       int retryNumber = 0;
-      while (retryNumber <= 5) {
-        if (retryNumber >= 1) {
-          waitForRetry(retryNumber);
+      while (result == null && loopNumber <= 5) {
+        if (loopNumber >= 1) {
+          waitForRetry(loopNumber, retryNumber);
         }
         try {
-          return readOrInsertIfMissing(mbi);
+          result = readOrInsertIfMissing(mbi, loopNumber > 0);
         } catch (PersistenceException ex) {
           final Throwable rootCause = Throwables.getRootCause(ex);
-          LOGGER.debug(
+          log.debug(
               "caught exception while caching MBI: retry={} class={} causeClass={}",
-              retryNumber,
+              loopNumber,
               ex.getClass().getSimpleName(),
               rootCause.getClass().getSimpleName());
           retryNumber += 1;
         }
+        if (result == null) {
+          loopNumber += 1;
+        }
       }
-      LOGGER.warn("unable to cache MBI after multiple tries, returning computed value");
-      return new Mbi(mbi, hasher.computeIdentifierHash(mbi));
+
+      if (loopNumber > 0) {
+        metrics.addMiss();
```

Comment:
Wouldn't it be a miss every time we get a null back in `readOrInsertIfMissing`?

Reply:
For the database cache it's only a miss if we had to add a record to the table for the new MBI.  Loop zero is just a read from the table so a match found during that loop isn't really a miss.

Reply:
As opposed to a non-database backed cache where if the value isn't already in memory its a miss.  

I guess we could track two different miss statistics but it seemed the database table is really our cache and the memory portion just an optimization.